src/plugins/intel_gpu/src/graph/eltwise.cpp

```diff
@@ -410,7 +410,11 @@ eltwise_inst::typed_primitive_inst(network& network, eltwise_node const& node) :
             auto base_pshape = input0_pshape;
             if (prim->broadcast_spec == ov::op::AutoBroadcastType::NUMPY &&
                 base_pshape.size() < input_pshape.size()) {
-                base_pshape.insert(base_pshape.begin(), input_pshape.size() - base_pshape.size(), 1);
+                if (use_new_shape_infer) {
+                    base_pshape.insert(base_pshape.begin(), input_pshape.size() - base_pshape.size(), 1);
+                } else {
+                    base_pshape.insert(base_pshape.end(), input_pshape.size() - base_pshape.size(), 1);
+                }
```

Comment:
`ov::op::AutoBroadcastType::NUMPY` means right-alignment. I don't understand why this change is needed.

Reply:
Update to align 4d for legacy static model (use_new_shape_infer false case)

Reply:
Update output shape of reshape op to align 4d for static case not using new_shape_infer